templates/deletion_flow.yaml

```diff
@@ -187,6 +187,11 @@ Resources:
           - Action: s3:GetObject*
             Effect: Allow
             Resource: !Sub arn:aws:s3:::${ManifestsBucket}/manifests/*
+          - Action:
+            - kms:Decrypt
+            - kms:GenerateDataKey*
+            Effect: Allow
+            Resource: !Sub arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/*
```

Comment:
can we scope this down at all? what about keys which aren't in the current account? Feels like we might need to parametrise this somehow

Reply:
Good point about cross-account! 
It could be messy unless I parametrise the arns in CFN I guess.

We have to consider that you still need to add the solution task arn in each key's policy as suggested by the policy generator, so I wonder if that is ok? I don't like wildcards but still, you need to go key by key anyways, would that be enough?

Reply:
Ok so I decided to parametrise the Key Arns, that allowed me to remove all wildcards in the policies as well as solving the cross-account issue.